syntax = "proto3";

package hostService;

service HostService {
    rpc Transmit(stream TransmitPacket) returns(stream TransmitResponse) {}
    rpc Processes(stream TransmitProcesses) returns(stream TransmitProcessesResponse) {}
}

message TransmitPacket {
    int64 MemoryUsed = 1;       // mem in kb
    int64 MemoryAvailable = 2;
    int64 MemoryTotal = 3;

    double CpuUsage = 4;    // % of the cores
    double UpTime = 5; // uptime in seconds

    string Version = 6; // version of the system
    string IP = 7; // inbound IP
    string DeviceID = 8;

    int64 InboundBandwithBytes = 9;
    int64 OutboundBandwithBytes = 10;
    int64 InboundBandwithPackets = 11;
    int64 OutboundBandwithPackets = 12;
    string hostName = 13;
}

message TransmitProcesses {
    repeated string Processes = 1;
}

message TransmitProcessesResponse {
    int32 FrequencyAdjustment = 2;
}

message TransmitResponse {
    bool DidInsert = 1;
    int32 FrequencyAdjustment = 2;
}
